editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -4818,6 +4818,18 @@ void CanvasItemEditor::remove_control_from_right_panel(Control *p_control) {
 	right_panel_split->remove_child(p_control);
 }
 
+void CanvasItemEditor::add_control_to_tool_drawer(const String p_name, const Ref<Texture2D> &p_icon, Control *p_control) {
+	tool_drawer->add_control_to_drawer(p_name, p_icon, p_control);
+}
+
+void CanvasItemEditor::remove_control_from_tool_drawer(Control *p_control) {
+	tool_drawer->remove_control_from_drawer(p_control);
+}
+
+void CanvasItemEditor::set_tool_drawer_visible(Control *p_control, bool p_visible) {
```

Comment:
This function sounds like it would toggle visibility of the whole drawer.

Now that I look at the names, they are a bit confusing. EditorToolDrawer seems to be a container for other drawers. It's like EditorToolDrawer and EditorToolDrawerGroup names are switched.

Maybe the drawer container should be EditorToolChest? (i.e. chest of drawers 🙃)

Reply:
I'm not entirely happy with the names of all 3 methods, 
```
add_control_to_tool_drawer
remove_control_from_tool_drawer
set_tool_drawer_visible
```

The first two mimic the names of other similar methods though. The last one, `set_tool_drawer_visible`, I could rename to `set_tool_drawer_enabled` maybe? Or maybe it's worth renaming all of them, but the problem is whether we are going to refer to individual panels as "drawers" or to the entire overlay.

Currently, the naming is as such: `EditorToolDrawer` is the name of the entire overlay, which is added over viewport or another container that needs it (say, visual script editor). `EditorToolDrawerSection` is the individual panel itself, which wraps around the control provided by the plugin developer. And `EditorToolDrawerGroup` is just a small helper for nice looking groups within each section/panel. The section > group dynamic follows the same of the inspector, so I think it's consistent... But once again, this boils down to what we are going to refer to as a "drawer".